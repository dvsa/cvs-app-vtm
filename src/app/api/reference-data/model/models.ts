export * from './reference-data.model';
