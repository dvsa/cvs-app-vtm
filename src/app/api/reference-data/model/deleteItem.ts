/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}