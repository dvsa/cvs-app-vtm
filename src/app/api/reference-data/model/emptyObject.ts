/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
