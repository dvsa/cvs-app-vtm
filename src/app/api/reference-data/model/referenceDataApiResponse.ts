/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { ReferenceDataApiResponseWithPagination } from './referenceDataApiResponseWithPagination';
import { ReferenceDataApiResponseWithoutPagination } from './referenceDataApiResponseWithoutPagination';

export type ReferenceDataApiResponse = ReferenceDataApiResponseWithoutPagination | ReferenceDataApiResponseWithPagination;