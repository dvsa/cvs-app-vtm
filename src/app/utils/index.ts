export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


