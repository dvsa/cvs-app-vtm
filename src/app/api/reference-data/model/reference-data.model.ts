import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
