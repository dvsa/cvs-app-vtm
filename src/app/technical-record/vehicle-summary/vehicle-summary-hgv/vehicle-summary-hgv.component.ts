import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { TechRecord } from '@app/models/tech-record.model';

@Component({
  selector: 'vtm-vehicle-summary-hgv',
  templateUrl: './vehicle-summary-hgv.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class VehicleSummaryHgvComponent implements OnInit {
  @Input() activeRecord: TechRecord;

  constructor() {}

  ngOnInit() {}
}
