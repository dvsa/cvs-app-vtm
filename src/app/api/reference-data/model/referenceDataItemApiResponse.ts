/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { EmptyObject } from './emptyObject';
import { ReferenceDataItem } from './referenceDataItem';

export type ReferenceDataItemApiResponse = ReferenceDataItem | EmptyObject;