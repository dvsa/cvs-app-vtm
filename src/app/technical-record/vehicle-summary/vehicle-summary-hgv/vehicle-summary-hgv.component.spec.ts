import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { VehicleSummaryHgvComponent } from './vehicle-summary-hgv.component';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { SharedModule } from '@app/shared/shared.module';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { TechRecord } from '@app/models/tech-record.model';

describe('VehicleSummaryHgvComponent', () => {
  let component: VehicleSummaryHgvComponent;
  let fixture: ComponentFixture<VehicleSummaryHgvComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [VehicleSummaryHgvComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(VehicleSummaryHgvComponent);
    component = fixture.componentInstance;
    component.activeRecord = {
      vehicleType: 'hgv',
      drawbarCouplingFitted: true,
      offRoad: false,
      speedLimiterMrk: true,
      tachoExemptMrk: true,
      euroStandard: '6',
      fuelPropulsionSystem: 'gas',
      numberOfWheelsDriven: 4,
      emissionsLimit: 23
    } as TechRecord;
  });

  it('should create view only with populated data', () => {
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });
});
