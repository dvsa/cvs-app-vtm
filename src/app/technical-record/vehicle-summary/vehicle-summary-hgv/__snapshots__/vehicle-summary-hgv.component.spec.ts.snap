// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleSummaryHgvComponent should create view only with populated data 1`] = `
<vtm-vehicle-summary-hgv
  activeRecord={[Function Object]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Speed limiter exempt 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-speedLimiterMrk"
      >
         Yes 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Tacho exempt 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-tachoExemptMrk"
      >
         Yes 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Euro standard 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-euroStandard"
      >
         6 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Fuel/propulsion system 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-fuelpropulsionsystem"
      >
         gas 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Number of wheels driven 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-numberOfWheelsDriven"
      >
         4 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Emission limit 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-emissionsLimit"
      >
         23 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Drawbar coupling fitted 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-drawbarCouplingFitted"
      >
         Yes 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Offroad vehicle 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-offRoad"
      >
         No 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Road friendly suspension 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-roadFriendly"
      >
         - 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
  </dl>
</vtm-vehicle-summary-hgv>
`;
